test/protobuf/encoder_test.exs

```diff
@@ -309,6 +310,15 @@ defmodule Protobuf.EncoderTest do
     end
   end
 
+  test "gives a warning for implicitly cast structs" do
+    warning =
+      ExUnit.CaptureIO.capture_io(:stderr, fn ->
```

Comment:
This makes this test not really async-friendly anymore. Can we turn it to `async: false` and leave a comment up there explaining why (and maybe a TODO to turn it back async when we remove this warning)?

Reply:
Great point, added!